### PR TITLE
Connect error handling

### PR DESCRIPTION
**Pull request by hdoordt**

### CODE REVIEW

**Comment by hdoordt** [comment]
All parts have been tested and are working. I just have not yet been able to test the situation as a whole

**Comment by Lucus16** [comment]
Please check if one of these needs to be updated:
- https://github.com/tweedegolf/veilig-bellen/blob/6cba294a2dec305281ec35268d2e0d493631b59d/backend/db.go#L99
- https://github.com/tweedegolf/veilig-bellen/blob/6cba294a2dec305281ec35268d2e0d493631b59d/backend/irma_poll.go#L69

**Comment by hdoordt** [comment]
The 

> Please check if one of these needs to be updated:
> 
>     * https://github.com/tweedegolf/veilig-bellen/blob/6cba294a2dec305281ec35268d2e0d493631b59d/backend/db.go#L9
>     * https://github.com/tweedegolf/veilig-bellen/blob/6cba294a2dec305281ec35268d2e0d493631b59d/backend/irma_poll.go#L69

The various states have not been updated, so these can be left as they are

File: backend/api.go [hunk 1]
```diff
@@ -252,12 +252,21 @@ func (cfg Configuration) handleSessionStatus(w http.ResponseWriter, r *http.Requ
...
+
 	secret, err := cfg.db.secretFromDTMF(dtmf)
+
 	if err == ErrNoRows {
 		http.Error(w, "session not found", http.StatusNotFound)
 	} else if err != nil {
 		log.Printf("failed to retrieve secret from dtmf: %v", err)
 		http.Error(w, "internal server error", http.StatusInternalServerError)
+	} else if callState == "unavailable" {
+		cfg.db.setStatus(secret, "UNAVAILABLE")
```
Comment (Lucus16) [inline backend/api.go:267]: This introduces the new `UNAVAILABLE` state right?

**Review by Lucus16 — APPROVED** [review]

File: backend/api.go (hunk 1 above)
Comment (hdoordt) [reply backend/api.go:267]: This state signifies that the call center was unavailable. It does not mean that the session is terminated, as the user could try calling again later.

**Pull request merged**